service/route_handlers.go

```diff
@@ -473,9 +482,29 @@ func getPossibleRequestIDs(r *http.Request, s *model.RancherSamlServiceProvider)
 			continue
 		}
 		claims := token.Claims.(jwt.MapClaims)
-		rv = append(rv, claims["id"].(string))
+		reqIDs = append(reqIDs, claims["id"].(string))
+		/*
+			HandleSamlLogin sets relayState per request, which contains the URI of the request. This URI has the redirectBackBase and
+			RedirectBackPath values. So we can get the redirectBackBase and redirectBackPath for the current login request from URI of
+			relayState
+		*/
+		redirectURI := claims["uri"].(string)
+		log.Debugf("redirectURI from claims: %v", redirectURI)
+		query, err := url.Parse(redirectURI)
+		if err != nil {
+			log.Errorf("Error in getting query params")
+			return []string{}, "", "", fmt.Errorf("Error in getting query params: %v", err)
+		}
+		parsedQuery, err := url.ParseQuery(query.RawQuery)
+		if err != nil {
+			log.Errorf("Error in parsing query params")
+			return []string{}, "", "", fmt.Errorf("Error in parsing query params: %v", err)
+		}
+		redirectBackBaseValue = parsedQuery.Get(redirectBackBase)
```

Comment:
Do you have fallback logic anywhere that will allow things to work if the redirect parameters are not on the URL? If not, I think you should add that

Reply:
Added the logic in HandleSamlAssertion